### PR TITLE
Update liboqs-codeowners per liboqs PR #1943

### PR DESCRIPTION
**Pull request by SWilson4**
See https://github.com/open-quantum-safe/liboqs/pull/1943.

Unconditionally, changes to `config.yaml` must
- [x] be approved by 2 members of the OQS TSC
- [x] not violate permissions documented in GOVERNANCE.md files for sub projects where such files exist

The following goals apply to changes to the file `config.yaml` with exceptions possible, as long as the rationale for the exception is documented by comments in the file:
- [x] all sub projects should be treated identically wrt roles & responsibilities as per the detailed list below
- [x] teams/team designations are to be used wherever possible; using personal GH handles should only be used in team definitions
- [x] Admin changes to the file must be documented by comments as to the rationale of the change

All the following conditions hold for permissions set in `config.yaml`:
-    sub project maintainers have admin rights on the sub projects
-    OQS and sub project release managers have maintainer rights on the sub projects but can themselves set/reset branch protection rules limiting write access to sensitive branches
-    sub project committers have write rights on all branches of the sub projects but can request branch protection rules limiting this
-    sub project contributors (incl. code owners) have write rights on all branches except main on those sub projects
-    OQS and sub project triage actors have triage rights on all branches of the sub projects
-    OQS maintainers and LF admins have admin rights on the organization (e.g., org-wide secret management) as well as maintenance rights on the team configurations

### CODE REVIEW

**Comment by SWilson4** [comment]
@open-quantum-safe/tsc The corresponding CODEOWNERS changes have been approved and are ready for merge in https://github.com/open-quantum-safe/liboqs/pull/1943 pending approval here. Per process, we need **two approvals** by TSC members here to proceed.

**Review by baentsch — APPROVED** [review]

**Review by Martyrshot — APPROVED** [review]

**Comment by SWilson4** [comment]
As per [request](https://github.com/open-quantum-safe/tsc/blob/main/oqs-status-meetings/2024-10-15%20-%20OQS%20status%20meeting%20-%20minutes.md#oqs-technical-steering-committee) from @christianpaquin, these are the people behind the affected GitHub handles for this PR:

- crockeea is Eric Crockett
- brian-jarvis-aws is Brian Jarvis
- cothan is Duc Tri Nguyen
- Martyrshot is Jason Goertzen
- praveksharma is Pravek Sharma
- ashman-p is Norman Ashley

**Review by dstebila — APPROVED** [review]

**Pull request merged**
